src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {

```

Comment:
![image](https://github.com/mate-academy/hibernate-user-service-hw/assets/12864128/c9bf831d-3385-4bc7-be84-d8fe9ec96b57)

Reply:
So you suggesting me to do all those operations and only AFTER them - check if user is not in DB? Why should i use resources for that, when existing approach uses less resources?

Reply:
Hashing isn't a particularly resource-intensive operation. Anyway, we don't need to getSalt of empty user optional
Also, while having a different message is more user-friendly, it's less secure, since we give clues to a potential attacker
```
    if (userCheck.isEmpty() || !passwordValidation(userCheck.get(), password)) {
        throw new AuthenticationException("Wrong login or password!");
    }
```